### PR TITLE
Improve documentation about storage.

### PR DESCRIPTION
**Pull request by beorn7**
Also add related FAQs (including the ZFS bug).

@juliusv

### CODE REVIEW

**Comment by beorn7** [comment]
Fixes https://github.com/prometheus/docs/issues/25 and also https://github.com/prometheus/prometheus/issues/484

File: content/docs/operating/storage.md [hunk 1]
```diff
@@ -5,10 +5,81 @@ nav_icon: database
...
+chunks are in use at the same time. You have to experiment a bit. The
+metrics `prometheus_local_storage_memory_chunks` and
+`process_resident_memory_bytes`, exported by the Prometheus server,
+will come in handy. As a rule of thumb, you should have at least three
+times more RAM available than needed by the memory chunks alone.
+
+LevelDB is essentially dealing with data on disk and relies on the
+disk caches of the operating system for optimal performance. However,
+it maintains in-memory caches, whose size you can configure for each
+index via the flags
```
Comment (juliusv) [inline content/docs/operating/storage.md]: How about "...via the following flags:" and then do a bulleted list?

File: content/docs/operating/storage.md [hunk 2]
```diff
@@ -5,10 +5,81 @@ nav_icon: database
...
 # Storage
 
-Prometheus stores its time series data under the directory specified by the flag
-`storage.local.path`. If you suspect problems caused by corruption in the
-database, or you simply want to erase the existing database, you can easily
-start fresh by deleting the contents of this directory:
+Prometheus has a sophisticated local storage subsystem. For indexes,
+it uses [LevelDB](https://github.com/google/leveldb). For the bulk
+sample data, it has its own custom storage layer, which organizes
+sample data in chunks of constant size (1024 bytes payload).
```
Comment (juliusv) [inline content/docs/operating/storage.md]: How about mentioning that each time series is one file: "which organizes sample data in chunks of constant size (...) and stores these chunks in a single file for each time series."

**Comment by juliusv** [comment]
:+1: besides comments

**Comment by beorn7** [comment]
Addressed all comments. Thanks.

**Pull request merged**